### PR TITLE
feat(ops): Consolidate Version Control

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Historically, binary version control in the monorepo was done via `.<binary>rc` files that each contain solely the version string for ced binary.

Recently, this has become used for an increasing number of binaries, greatly increasing the number of top-level hidden monorepo source files.

This PR consolidates checked in binary versions in a singular top-level `versions.json` file where binaries are mapped to their version string or commit hash (for eg foundry).

The hidden `rc` files are now deleted and referencing code has been updated to read and write to the version control `versions.json` file.

Since this impacts developer use across the monorepo, would appreciate meticulous review and critiques!

### CODE REVIEW

**Comment by refcell** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #8920** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8920?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8920?utm_source=stack-comment).

**Review by Inphi — COMMENTED** [review]
It's a cool idea. I think this issue could also be solved by moving the rc files into a subdirectory so as to not pollute the top level. Avoids introducing new code or requiring JSON for something this basic. wydt?

**Review by trianglesphere — COMMENTED** [review]
I like this cleanup. The two things I see are that we need to test the CI builder & there's a merge conflict with develop.

**Comment by refcell** [comment]
> It's a cool idea. I think this issue could also be solved by moving the rc files into a subdirectory so as to not pollute the top level. Avoids introducing new code or requiring JSON for something this basic. wydt?

Not a bad way to go at all. I think that's a perfectly reasonable alternative we can do if others prefer!
I still lean toward this solution because to me it's cleaner to have a single top-level file that handles versioning as opposed to navigating into yet another sub directory. Plus the additional code is minimal enough to where the tradeoff seems worthwhile

File: Makefile [hunk 1]
```diff
@@ -197,6 +197,6 @@ install-geth:
 	./ops/scripts/geth-version-checker.sh && \
 	 	(echo "Geth versions match, not installing geth..."; true) || \
  		(echo "Versions do not match, installing geth!"; \
- 			go install -v github.com/ethereum/go-ethereum/cmd/geth@$(shell cat .gethrc); \
+ 			go install -v github.com/ethereum/go-ethereum/cmd/geth@$(shell cat versions.json | jq -r '.geth'); \
  			echo "Installed geth!"; true)
```
Comment (tynes) [inline Makefile:205]: a neckbeard would say this is a useless cat because you can just do `jq -r .geth < versions.json`

**Comment by tynes** [comment]
I'm generally in favor of this approach

File: op-bindings/bindgen/utils.go [hunk 2]
```diff
@@ -151,13 +152,12 @@ func genContractBindings(logger log.Logger, monorepoRootPath, abiFilePath, bytec
 		abigenVersion := bytes.Trim(versionBuf.Bytes(), "\n")
 
 		// Fetch expected abigen version (format: vX.Y.Z)
```
Comment (tynes) [inline op-bindings/bindgen/utils.go:154]: I think its weird that `bindgen` itself is coupled to the version serialization but its nbd

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
Other instances of useless cat but not a problem

**Pull request merged**